lib/redis_client/config.rb

```diff
@@ -151,6 +151,7 @@ def build_connection_prelude
         if @db && @db != 0
           prelude << ["SELECT", @db.to_s]
         end
+        prelude.map(&:freeze)
```

Comment:
```suggestion
        prelude.each(&:freeze)
```

Reply:
But also it's freezing strings we received, so we probably should use `String#-@` rather than freeze, or `str.frozen? ? str : str.dup.freeze`.

Reply:
Also:

> While the prelude array was frozen, the strings themselves were not.

`prelude` is an array of arrays of strings, so this actually freeze the second level of arrays, not the strings.

Reply:
Yeah, I just realized my mistake about the array of arrays. Thanks!